api/base/utils.py

```diff
@@ -19,17 +19,32 @@ def absolute_reverse(view_name, query_kwargs=None, args=None, kwargs=None):
     return url
 
 
-def get_object_or_404(model_cls, query_or_pk):
+def get_object_or_error(model_cls, query_or_pk, display_name=None):
+
+    display_name = display_name or None
+
     if isinstance(query_or_pk, basestring):
         query = Q('_id', 'eq', query_or_pk)
     else:
         query = query_or_pk
+
     try:
-        return model_cls.find_one(query)
+        obj = model_cls.find_one(query)
+        if getattr(obj, 'is_deleted', False) is True:
+            if display_name is None:
+                raise Gone
+            else:
+                raise Gone(detail='The requested {name} is no longer available.'.format(name=display_name))
+        if getattr(obj, 'is_active', None) is False:
```

Comment:
What if `is_active` is never set, though? Or users that were created before the `is_active` field was added? I'd expect mongo/odm to return `None`.